lua/vgit/features/screens/ProjectDiffScreen/init.lua

```diff
@@ -111,13 +126,43 @@ function ProjectDiffScreen:stage_hunk()
   self:render(function()
     local has_unstaged = false
     self.status_list_view:each_status(function(status, entry_type)
-      if entry_type == 'unstaged' and status.filename == entry.status.filename then has_unstaged = true end
-    end)
-    self:move_to(function(status, entry_type)
-      if has_unstaged and entry_type == 'staged' then return false end
-      return status.filename == entry.status.filename
+      if entry_type == 'unstaged' and status.filename == filename then
+        has_unstaged = true
+      end
     end)
+
+    if has_unstaged then
+      self:move_to(function(status, entry_type)
+        return status.filename == filename and entry_type == 'unstaged'
+      end)
+    elseif next_file then
+      self:move_to(function(status, entry_type)
+        return status.filename == next_file and entry_type == 'unstaged'
+      end)
+    else
+      local found = self:move_to(function(_, entry_type)
+        return entry_type == 'unstaged'
+      end)
+      if not found then
+        self:move_to(function(status)
+          return status.filename == filename
+        end)
+      end
```

Comment:
The condition for `next_file` should cover moving to the next unstaged file (when there are NO other unstaged files and we just staged the last unstaged hunk).

Let me know otherwise, then we can also apply it to all other similar methods.

```suggestion
      self:move_to(function(status)
        return status.filename == filename
      end)
```